dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -1426,6 +1448,14 @@ Sedp::update_topic_qos(const RepoId& topicId, const DDS::TopicQos& qos)
 DDS::ReturnCode_t
 Sedp::remove_publication_i(const RepoId& publicationId, LocalPublication& pub)
 {
+  if (!(spdp_.available_builtin_endpoints() & (DISC_BUILTIN_ENDPOINT_PUBLICATION_ANNOUNCER
```

Comment:
I think we don't need this block (and the corresponding one in `remove_subscription_i`) because the entity being removed couldn't have been added in the first place.